### DISCUSSION TITLE
working-with-git-in-jetbrains-rider/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#123] [Announcements]
# working-with-git-in-jetbrains-rider/

In this post, I describe how I work day-to-day with Git using JetBrains' Rider, working exclusively with keyboard shortcuts, but with the benefits of a GUI

https://andrewlock.net/working-with-git-in-jetbrains-rider/

**Comment by silkfire** [comment]
This is a bit off-topic, but I am experiencing a very annoying thing with the autocompletion functionality in Rider. The first time I either autocomplete a member name (Ctrl+Space) or show Parameter Info (Ctrl+Shift+Space) in a specific part of the code, the tooltip is shown for a split second and then disappears. On the following or the third attempt, the tooltip stays there. Is this a know bug or maybe even a "feature"?

**Comment by thomaslevesque** [comment]
Nice! I've been using Rider for a while, but for Git I mostly stuck to the command line so far, because I usually don't like the built-in Git experience in IDEs. I'll give it a try!

BTW, your link to "Think like a Git is broken". The correct link is https://think-like-a-git.net/

**Comment by khalidabuhakmeh** [comment]
👋 Hi @silkfire, I haven't experienced that issue. If you can recreate it reliably, please file an issue with the JetBrains Rider issue tracker. Also, try our EAPs, as they're generally stable and have bug fixes, but your mileage may vary.

**Comment by mikoskinen** [comment]
Thanks, this was a well timed post :) I've been full-time Rider user for about 3.5 years now mainly because of the performance: Notepad vs Word is how I often describe the performance between Rider and Visual Studio.

For the past 10 years or so I've also used TortoiseGit as my go to Git GUI but recently I've been transitioning to Mac where TortoiseGit isn't an option. I went through the often mentioned Git clients but ended up using Rider's built-in functionalities. 

Thanks again for good usage tips on this subject.
